xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,478 @@
+    }
+    if (infoA.channels != infoB.channels)
+    {
+      return infoA.channels < infoB.channels;
+    }
+    if (infoA.isDefault != infoB.isDefault)
+    {
+      return infoA.isDefault < infoB.isDefault;
+    }
+    if (infoA.isForced != infoB.isForced)
+    {
+      return infoA.isForced < infoB.isForced;
+    }
+    if (infoA.isOriginal != infoB.isOriginal)
+    {
+      return infoA.isOriginal < infoB.isOriginal;
+    }
+    if (infoA.isHearingImpaired != infoB.isHearingImpaired)
+    {
+      return infoA.isHearingImpaired < infoB.isHearingImpaired;
+    }
+    if (infoA.isVisualImpaired != infoB.isVisualImpaired)
+    {
+      return infoA.isVisualImpaired < infoB.isVisualImpaired;
+    }
+    if (infoA.bitrate != infoB.bitrate)
+    {
+      return infoA.bitrate < infoB.bitrate;
+    }
+    return infoA.samplerate < infoB.samplerate;
+  }
+};
+
+struct SortComparerStreamSubtitle
+{
+  bool operator()(const std::pair<int, SubtitleStreamInfoExt>& a,
+                  const std::pair<int, SubtitleStreamInfoExt>& b)
+  {
+    const SubtitleStreamInfoExt& infoA = a.second;
+    const SubtitleStreamInfoExt& infoB = b.second;
+
+    if (infoA.languageDesc != infoB.languageDesc)
+    {
+      return infoA.languageDesc < infoB.languageDesc;
+    }
+    if (infoA.isDefault != infoB.isDefault)
+    {
+      return infoA.isDefault < infoB.isDefault;
+    }
+    if (infoA.isForced != infoB.isForced)
+    {
+      return infoA.isForced < infoB.isForced;
+    }
+    if (infoA.isHearingImpaired != infoB.isHearingImpaired)
+    {
+      return infoA.isHearingImpaired < infoB.isHearingImpaired;
+    }
+    if (infoA.isVisualImpaired != infoB.isVisualImpaired)
+    {
+      return infoA.isVisualImpaired < infoB.isVisualImpaired;
+    }
+    return infoA.codecName < infoB.codecName;
+  }
+};
+
+} // unnamed namespace
+
+void KODI::VIDEO::GUILIB::OpenDialogSelectVideoStream()
+{
+  CGUIDialogSelect* dialog{CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+      WINDOW_DIALOG_SELECT_VIDEO_STREAM)};
+  if (!dialog)
+  {
+    CLog::LogF(LOGERROR, "Unable to get WINDOW_DIALOG_SELECT_VIDEO_STREAM dialog instance");
+    return;
+  }
+
+  auto& components = CServiceBroker::GetAppComponents();
+  auto appPlayer = components.GetComponent<CApplicationPlayer>();
+  const int streamCount = appPlayer->GetVideoStreamCount();
+  const int selectedIndex = appPlayer->GetVideoStream();
+
+  std::vector<std::pair<int, VideoStreamInfoExt>> streams;
+  streams.reserve(streamCount);
+
+  // Collect all streams
+  for (int i = 0; i < streamCount; ++i)
+  {
+    VideoStreamInfoExt info;
+    appPlayer->GetVideoStreamInfo(i, info);
+
+    info.isDefault = info.flags & StreamFlags::FLAG_DEFAULT;
+    info.isForced = info.flags & StreamFlags::FLAG_FORCED;
+    info.isHearingImpaired = info.flags & StreamFlags::FLAG_HEARING_IMPAIRED;
+    info.isVisualImpaired = info.flags & StreamFlags::FLAG_VISUAL_IMPAIRED;
```

Comment:
i reworked code in better way, also changed "index" names to "id" since i realized that is called "id" also on others kodi core parts, removed also the needs of "pairs"